src/engraving/rendering/dev/chordlayout.cpp

```diff
@@ -3276,7 +3325,9 @@ void ChordLayout::fillShape(const Chord* item, ChordRest::LayoutData* ldata)
 //      if (_tremolo)
 //            shape.add(_tremolo->shape().translated(_tremolo->pos()));
     for (Note* note : item->notes()) {
-        shape.add(note->shape().translate(note->pos()));
+        Shape noteShape = note->shape();
+        noteShape.remove_if([&](ShapeElement& s) { return !s.item()->addToSkyline() && !s.item()->isNote(); });
```

Comment:
Thank you @XiaoMigros, this is good work! Just one thing: do I understand correctly that the purpose of this line is to remove the invisible dots from the Note shape? If so, it should be done inside the TLayout::fillNoteShape function (where you can just skip the invisible dots). It's both more efficient and more tidy.

Reply:
Thanks, that's much more elegant! Has been taken care of :)